File: LojaDeLivros/app/Models/Book.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Book extends Model
{
    /** @use HasFactory<\Database\Factories\BookFactory> */
    use HasFactory;

    protected $fillable = [
        'title',
        'author',
        'description',
        'price',
        'image',
        'category_id',
        'owner_id',
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function owner()
    {
        return $this->belongsTo(User::class, 'owner_id');
    }
}

File: mobile_app/lib/models/book.dart
import 'category.dart';

class Book {
  final int id;
  final String title;
  final String author;
  final String description;
  final double price;
  final String? imageUrl;
  final Category? category;

  Book({
    required this.id,
    required this.title,
    required this.author,
    required this.description,
    required this.price,
    this.imageUrl,
    this.category,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      author: json['author'],
      description: json['description'] ?? '',
      price: double.parse(json['price'].toString()),
      imageUrl: json['image_url'],
      category: json['category'] != null ? Category.fromJson(json['category']) : null,
    );
  }
}
